mpqp/noise/noise_model.py

```diff
@@ -277,25 +277,103 @@ def to_other_language(
 
 
 class BitFlip(NoiseModel):
-    """3M-TODO"""
+    """Class representing the bit flip noise channel, which flips the state of
+    a qubit with a certain probability. It can be applied to single-qubit gates
+    and depends on a single parameter (probability or error rate).
 
-    # def __init__(
-    #     self,
-    #     proba: Union[float, Expr],
-    #     targets: List[int],
-    #     dimension: int = 1,
-    #     gates: List[Gate] = None):
+    Args:
+        prob: Bit flip error probability or error rate (must be within [0, 0.5]).
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise. Only single-qubit gates are valid.
 
-    #     super().__init__(proba, targets, dimension, gates)
+    Raises:
+        ValueError: When the probability is outside of the expected interval [0, 0.5].
+        ValueError: When no target qubits are specified.
+        ValueError: When the specified gates are not single-qubit gates.
 
-    def to_kraus_representation(self) -> KrausRepresentation:
-        # generate Kraus operators for bit flip noise
-        # kraus_operators = [
-        #     np.sqrt(1 - self.proba) * np.array([[1, 0], [0, 1]]),  # Identity
-        #     np.sqrt(self.proba) * np.array([[0, 1], [1, 0]])      # Bit flip
-        # ]
-        # return KrausRepresentation(kraus_operators)
-        ...
+    Examples:
+        >>> circuit = QCircuit([H(i) for i in range(3)])
+        >>> bf1 = BitFlip(0.1, [0])
```

Comment:
Bitflip was not added to the all.py and __init__.py of mpqp.noise ?

Reply:
Added.

Reply:
Apparently not @MoHermes